### PR TITLE
Provide a starting point for terminology

### PR DESCRIPTION
**Pull request by appgurueu**
Some terms are used in too many ways, so I started a terminology documentation. Feel free to add terms by editing this PR.

### CODE REVIEW

File: doc/terminology.adoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+= Terminology
+
+== Lua
+
+Class:: A metatable that sets `__index`
+Instance:: An object that has a Class set as it's metatable
+Method:: A function entry in the metatable, callable using `instance:method(args)`
+List:: A table using only the list part (that is, only keys from `1` to `#table`)
+Integer:: A Lua number without fractional part (will be stored as a double though)
```
Comment (Desour) [inline doc/terminology.adoc:9]:
```suggestion
Integer:: A Lua number without fractional part
```
Actually luajit can store numbers as ints if it compiles. (Can be seen with jit.dump in for loops. I think I've already managed to cause an overflow at some point, but I can't reproduce at the moment.)

File: doc/terminology.adoc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+= Terminology
+
+== Lua
+
+Class:: A metatable that sets `__index`
```
Comment (Desour) [inline doc/terminology.adoc:5]: Classes are rather an abstract construct. Metatables are used to implement classes, but not the classes themselves.
Maybe just link to wikipedia here.

**Comment by benrob0329** [comment]
Is this actually all that useful as a general document? It's using terms which Lua modders won't encounter when working with the API (Lua 5.1/JIT doesn't have classes or integers in a user-facing way, just `table`s/`userdata` and `number`s).

We should aim to use terms which people will actually encounter, rather than those they won't, an explain these terms where they are used. Having some macros for the explanation blurbs might be a better idea.

**Comment by appgurueu** [comment]
The entire point of this is to explain otherwise ambiguous terms like "integer" (which might be misunderstood as an actual "int" datatype while it's still Lua's "number" datatype in the end), not to repeat Lua(JIT) terminology.

**Comment by benrob0329** [comment]
It'd probably be better to have a full article on Data Types then, rather than just a "terminology" doc. I'm willing to help write that, if you want.

**Pull request closed without merging**